driver_go18_test.go

```diff
@@ -247,12 +293,33 @@ func TestContextCancelExec(t *testing.T) {
 		dbt.mustExec("CREATE TABLE test (v INTEGER)")
 		ctx, cancel := context.WithCancel(context.Background())
 
-		// Delay execution for just a bit until db.ExecContext has begun.
+		var longProcess *dbProcess
+		longQuery := "INSERT INTO test VALUES (SLEEP(1))"
+
+		// Delay execution for just a bit until db.ExecContext has begun and get long process ID
+		defer time.AfterFunc(100*time.Millisecond, func() {
+			processList, err := getProcesslist(dbname, dbt.db)
+			if err != nil {
+				dbt.Errorf("failed to get mysql process list %v", err)
+			}
+			longProcess = getQueryProcess(processList, longQuery)
+		}).Stop()
+
+		// Delay execution for just a bit until db.ExecContext has begun and cancel context
 		defer time.AfterFunc(100*time.Millisecond, cancel).Stop()
 
+		// Delay execution for just a bit until db.ExecContext is cancelled and check that the process has been killed
+		defer time.AfterFunc(150*time.Millisecond, func() {
```

Comment:
find the query id again, check every 100 ms, with timeout 5sec, if timeout exceeds and query does not kill, the test fails.

Reply:
in summary:
15 sec for execution query:
5 sec to start query.
5 sec to kill query.
I think this will be enough in any case.